mypy/server/update.py

```diff
@@ -1027,6 +1029,10 @@ def key(node: FineGrainedDeferredNode) -> int:
         if graph[module_id].type_checker().check_second_pass():
             more = True
 
+    graph[module_id].detect_possibly_undefined_vars()
+    graph[module_id].generate_unused_ignore_notes()
+    graph[module_id].generate_ignore_without_code_notes()
```

Comment:
I think that it may be more correct to only perform these after the end of a build increment, after we've propagated all changes. After a call of `reprocess_nodes` we might still be in the middle of processing some files, and some errors could still be not generated yet. So maybe we'd need to keep track of all the files that have processed nodes, and call these functions at the end?

@ilevkivskyi What do you think?

Reply:
I don't know TBH, I thought we clear those "intermediate" errors. So I guess we may clear them for some reason? If yes, this should be fixed. But if we fixing that is hard, then yes, we can delay generating these errors.

Reply:
Hi both -- thank you for your comments on this.

Apologies for not addressing your points before.

I've re-opened this PR (with some minor changes to address a regression) in https://github.com/python/mypy/pull/15043, and I'd appreciate your feedback there. In particular, I'd really welcome your thoughts on what should be done to address the comments you left here.

Reply:
FWIW, I have removed these three lines from the new PR.